collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2394,7 +2390,18 @@ public void denied() {
                 }
 
                 if (formController.getInstanceFile() == null) {
-                    createInstanceDirectory(formController);
+                    FormInstanceFileCreator formInstanceFileCreator = new FormInstanceFileCreator(
+                            storagePathProvider,
+                            System::currentTimeMillis
+                    );
+
+                    File instanceFile = formInstanceFileCreator.createInstanceFile(formPath);
+                    if (instanceFile != null) {
+                        formController.setInstanceFile(instanceFile);
+                    } else {
+                        showFormLoadErrorAndExit(getString(R.string.form_instance_file_creation_error));
```

Comment:
Could you also do a `Timber.e("Error creating form instance file")` so we can see it in Crashlytics, keep an eye on how common it is, and possibly learn something from the logs?

Reply:
Yeah good call!